ios/browser/api/history/brave_history_observer.mm

```diff
@@ -0,0 +1,165 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/history/brave_history_observer.h"
+#include "brave/ios/browser/api/history/brave_history_api.h"
+
+#include <memory>
+
+#include "base/check.h"
+#include "base/compiler_specific.h"
+#include "base/strings/sys_string_conversions.h"
+
+#include "components/history/core/browser/history_service.h"
+#include "components/history/core/browser/history_types.h"
+#include "components/history/core/browser/history_service_observer.h"
+
+#import "net/base/mac/url_conversions.h"
+#import "url/gurl.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+@interface IOSHistoryNode (Private)
+- (instancetype)initWithURL:(NSURL*)url
+                      title:(NSString* _Nullable)title
+                  dateAdded:(NSDate* _Nullable)dateAdded;
+@end
+
+namespace brave {
+namespace ios {
+class HistoryServiceListener : public history::HistoryServiceObserver {
```

Comment:
Rename to `HistoryServiceListenerIOS`

Reply:
Oh I just realized the class you're inheriting from is `HistoryServiceObserver` not listener… In that case since its a local file and its not exposed its probably fine to stay as `HistoryServiceListener`... If you do rename, make it `HistoryServiceObserverIOS`

Reply:
`HistoryServiceObserverIOS` renaming done.